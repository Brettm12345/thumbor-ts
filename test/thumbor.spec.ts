import Thumbor from '../src';

describe('thumbor', () => {
  const image = Thumbor({
    serverUrl: 'https://thumbor',
    imagePath: 'example.png'
  });

  it('Generates urls', () => {
    expect(image.buildUrl()).toMatchSnapshot();
  });

  it('Removes duplicates', () => {
    expect(
      image
        .grayscale()
        .grayscale()
        .grayscale()
        .buildUrl()
    ).toMatchSnapshot();
  });

  it('Works with resize', () => {
    expect(image.resize(30, 30).buildUrl()).toMatchSnapshot();
  });

  it('Works with crop', () => {
    expect(image.crop(10, 10, 30, 30).buildUrl()).toMatchSnapshot();
  });

  it('Properly flips images', () => {
    expect(
      image.resize(30, 30, true, true).buildUrl()
    ).toMatchSnapshot();
  });

  it('Works with filter', () => {
    expect(image.quality(80).buildUrl()).toMatchSnapshot();
  });

  it('Works with many filters', () => {
    expect(
      image
        .quality(80)
        .autoJpg()
        .format('webp')
        .buildUrl()
    ).toMatchSnapshot();
  });

  it('Works with security keys', () => {
    expect(
      image
        .setSecurityKey('12345')
        .resize(30, 30)
        .buildUrl()
    ).toMatchSnapshot();
  });
});
