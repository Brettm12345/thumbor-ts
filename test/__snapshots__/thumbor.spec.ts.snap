// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`thumbor Generates urls 1`] = `"http://localhost/unsafe/react-day-picker.png"`;

exports[`thumbor Removes duplicates 1`] = `"http://localhost/unsafe/filters:grayscale()/react-day-picker.png"`;

exports[`thumbor Works with crop 1`] = `"http://localhost/unsafe/10x10:30x30/react-day-picker.png"`;

exports[`thumbor Works with filter 1`] = `"http://localhost/unsafe/filters:quality(80)/react-day-picker.png"`;

exports[`thumbor Works with many filters 1`] = `"http://localhost/unsafe/filters:quality(80):autoJpg():format(webp)/react-day-picker.png"`;

exports[`thumbor Works with resize 1`] = `"http://localhost/unsafe/30x30/react-day-picker.png"`;

exports[`thumbor Works with security keys 1`] = `"http://localhost/55r7U796iucpURlC4L2L5bL6aAk=/30x30/react-day-picker.png"`;
